flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/StateWithExecutionGraph.java

```diff
@@ -286,7 +286,7 @@ CompletableFuture<String> triggerCheckpoint() {
         logger.info("Triggering a checkpoint for job {}.", jobID);
 
         return checkpointCoordinator
-                .triggerCheckpoint(false)
+                .triggerCheckpointFromCheckpointThread(false)
```

Comment:
The old method can now be made private. Then I'd named it `triggerCheckpointInternal` and name the new one (which uses `timer.execute`) `triggerCheckpoint`. 
The caller of the public method shouldn't chose which thread to use IMO.
WDYT?

Reply:
I thought about that. But I was afraid of the number of tests that use the synchronous triggering checkpoints. But now, as I see, perhaps, not all of them really require synchronous checkpoint. So I will take a look at how is difficult to change another tests.